testsuite/tests/mgc/reconciliation/test_same_target.py

```diff
@@ -0,0 +1,49 @@
+"""Tests that DNSPolicy/TLSPolicy is rejected when the Gateway already has a policy of the same kind"""
+
+import pytest
+from openshift_client import selector
+
+from testsuite.policy.dns_policy import DNSPolicy
+from testsuite.tests.mgc.reconciliation import tls_policy
+
+pytestmark = [pytest.mark.mgc]
+
+
+@pytest.fixture(scope="module")
+def base_domain(hub_openshift):
+    """Returns preconfigured base domain"""
+    zone = selector("managedzone/aws-mz", static_context=hub_openshift.context).object()
+    return zone.model["spec"]["domainName"]
+
+
+@pytest.mark.parametrize(
+    "create_cr", [pytest.param(DNSPolicy.create_instance, id="DNSPolicy"), pytest.param(tls_policy, id="TLSPolicy")]
+)
+def test_two_policies_one_gw(request, create_cr, hub_gateway, client, blame, module_label):
+    """Tests that DNSPolicy/TLSPolicy is rejected when the Gateway already has a DNSPolicy"""
+
+    def two_dns_policies_error(policy: DNSPolicy):
```

Comment:
Only DNSPolicy here in the function name and the type:
```suggestion
    def two_policies_error(policy: DNSPolicy | TLSPolicy):
```

Reply:
Removed the type altogether, it didn´t  serve any purpose there